src/rules/normal_mean_precision/mean.jl

```diff
@@ -12,4 +12,7 @@ end
 # --------------------------------- #
 @rule NormalMeanPrecision(:μ, Marginalisation) (q_out::Any, q_τ::Any) = NormalMeanPrecision(mean(q_out), mean(q_τ))
 
-@rule NormalMeanPrecision(:μ, Marginalisation) (m_out::NormalMeanPrecision, q_τ::Any) = NormalMeanPrecision(mean(m_out), cholinv( cov(m_out) + cholinv(mean(q_τ)) ))
\ No newline at end of file
+@rule NormalMeanPrecision(:μ, Marginalisation) (m_out::UnivariateNormalDistributionsFamily, q_τ::Any) = begin 
+    m_out_mean, m_out_cov = mean_cov(m_out)
+    return NormalMeanPrecision(m_out_mean, cholinv(m_out_cov + cholinv(mean(q_τ)) ))
```

Comment:
Why do we return a precision matrix here? A covariance matrix would make more sense as this requires one less inversion. If an inversion is required later on, then this can still be performed.

Reply:
That is true. In the beginning of this package idea was that a node should always return the same parametrisation as a message. However, I don't really think it is that important and we can return `NormalMeanVariance` here indeed. What do you think @albertpod? I think we have the same for multivariate case as well (extra inversion just to return a precision matrix).

Reply:
It's hard to tell which parametrization is preferable in advance.
The rule of thumb is to return a message that has fewer inversions, so in this example, `NormalMeanVariance` is a better option.
On the other hand, we know that the multiplication of Gaussians can be optimized by using their "natural parameterization". So if a Gaussian rule can be represented by weighted-mean-precision (wmp) with fewer inversions, then better to do so. `ReactiveMP.jl` casts the factors of products of two Gaussians to wmp.

Reply:
I think we should not assume in advance that we are going to do a multiplication of Gaussians. Even if we do, it doesn't save us even a single inversion:

- Strategy 1: Single inverse in rule - no inversions in prod
- Strategy 2: No inversions in rule - Single inverse in prod

In that sense, I thing `NormalMeanVariance` might be better if we don't do `prod` after and also doesn't affect performance if we do `prod` right after.